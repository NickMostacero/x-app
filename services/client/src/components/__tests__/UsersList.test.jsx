import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import UsersList from '../UsersList';

const users = [
    {
        'active': true,
        'email': 'nickmostacero@upeu.edu.pe',
        'id': 1,
        'username': 'ldragons'
    },
    {
        'active': true,
        'email': 'pibex.g.m@hotmail.es',
        'id': 2,
        'username': 'brayan'
    }
];

test('UsersList renders properly', () =>{
    const wrapper = shallow(<UsersList users={users}/>);
    const element = wrapper.find('h4');
    expect(element.length).toBe(2);
    expect(element.get(0).props.children).toBe('ldragons');
});     

test('UsersList renders a snapshot', () =>{
    const tree = renderer.create(<UsersList users={users}/>).toJSON();
    expect(tree).toMatchSnapshot();
});  