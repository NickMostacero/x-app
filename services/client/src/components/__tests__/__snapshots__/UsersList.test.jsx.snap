// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders a snapshot 1`] = `
<div>
  <h4
    className="box title is-4"
  >
    ldragons
  </h4>
  <h4
    className="box title is-4"
  >
    brayan
  </h4>
</div>
`;
